test/test_device_cacophony_index.py

```diff
@@ -0,0 +1,124 @@
+                )
+
+    def now(self):
+        return self._now
+
+    def get_device(self):
+        return self._device
+
+    def get_user(self):
+        return self._user
+
+    def test_get_cacophony_index(self, helper):
+        admin = helper.admin_user()
+        johnny = self.get_user()
+        print("If a user Johnny wants to access the cacophony index histogram for a device")
+        cacophony_index = johnny.get_cacophony_index_for_device(self.get_device())
+        assert cacophony_index["cacophonyIndex"] == 11.5
+        cacophony_index_admin = admin.get_cacophony_index_for_device(self.get_device())
+        assert cacophony_index_admin["cacophonyIndex"] == 11.5
+
+        now = self.now()
+        # We only have 1 day of index values in the DB, so get only a window of 12 hours
+        print(f"Get cacophony index from now ({js_iso_format_with_utc(now)}, {now}), going back 12 hours")
+        cacophony_index_twelve_hours = johnny.get_cacophony_index_for_device(
+            self.get_device(), js_iso_format_with_utc(now), 12
+        )
+        assert cacophony_index_twelve_hours["cacophonyIndex"] == 5.5
+
+        print(f"Get cacophony index from now ({js_iso_format_with_utc(now)}), going back 0 hours")
+        cacophony_index_zero_window = johnny.get_cacophony_index_for_device(
+            self.get_device(), js_iso_format_with_utc(now), 0
+        )
+        assert cacophony_index_zero_window["cacophonyIndex"] is None
+
+        now_minus_twelve_hours = js_iso_format_with_utc(now - timedelta(hours=12))
+        print(f"Get cacophony index from 12 hours ago ({now_minus_twelve_hours}), going back 12 hours")
+        cacophony_index_from_twelve_hours_ago = johnny.get_cacophony_index_for_device(
+            self.get_device(), now_minus_twelve_hours, 12,
+        )
+        assert cacophony_index_from_twelve_hours_ago["cacophonyIndex"] == 17.5
+
+        # Make sure only the correct users can see the device cacophony-index.
+        random_user = helper.given_new_user(self, "dis_random_guy")
+        try:
+            random_user.get_cacophony_index_for_device(self.get_device())
+        except AuthorizationError:
+            pass
+
+    def test_get_cacophony_index_histogram(self, helper):
+        admin = helper.admin_user()
+        johnny = self.get_user()
+        cacophony_index = johnny.get_cacophony_index_histogram_for_device(self.get_device())
+        assert len(cacophony_index["cacophonyIndex"]) == 24
+        cacophony_index_admin = admin.get_cacophony_index_histogram_for_device(self.get_device())
+        assert len(cacophony_index_admin["cacophonyIndex"]) == 24
+
+        now = self.now()
+        cacophony_index_from_twelve_hours_ago = johnny.get_cacophony_index_histogram_for_device(
+            self.get_device(), js_iso_format_with_utc(now - timedelta(hours=12)), 12,
+        )
+        assert len(cacophony_index_from_twelve_hours_ago["cacophonyIndex"]) == 12
```

Comment:
could assert that the index values are all over 0 or something

Reply:
> 
> 
> Looks good a few bits and pieces that should be tidied up.
> Also any reason for not using the Sequelize query format? we are using it everywhere, so good to use it to keep things consistent.

The main reason I am not using Sequelize query format is that I have not the slightest idea how I would start to express a query like this in sequelize format.  Do you know how you'd do it?